src/createDispatcher.js

```diff
@@ -50,19 +56,27 @@ export default function createDispatcher(opts = {}) {
 
   const logging = parseOpts(opts.logging)
   if (logging.agendas) {
+    // Seems like this means that middlewares are effectively an attachment on dispatchers
     logAgendas(dispatcher)
   }
 
   function next(agenda) {
     const obs = agenda
       .subscribeOn(scheduler)
-      .publishReplay()
+      .publishReplay() // Would this make all my actions replayable?
+                       // What if they are expensive or can't be performed again?
+                       // I definitely need to look up publishReplay when I have internet again
       .refCount()
```

Comment:
Also, I was experimenting with the reducing logic being outside of the reduce method, so that it's cleanly separated. Writing the logic separated from the user method is something that I need to explore. Once we've got something like that, we can get rid of publish replay